pkg/cloudprovider/aws/apis/v1alpha1/provider_defaults.go

```diff
@@ -75,3 +76,10 @@ func (c *Constraints) defaultSecurityGroups(clusterName string) {
 	}
 	c.SecurityGroupSelector = map[string]string{fmt.Sprintf(ClusterDiscoveryTagKeyFormat, clusterName): "*"}
 }
+
+func (c *Constraints) defaultInstanceProfile(defaultInstanceProfile string) {
```

Comment:
This will cause some drift if the karpenter controller is restarted w/ a new default-instance-profile. Any provisioner created on the previous install will stay on the previous value, which may be surprising. 

I'd probably just leave any defaulting out and just use the value if it happens to be set, or fallback to GetOptions(ctx). WDYT?

Reply:
Ah! Good catch there. I'll admit I hadn't thought of that as I'm still fairly new to controller development. I'll send something your way in just a few mins 👍

Reply:
So yeah... there is a problem if the `spec.provider` is empty. While I can obviously skip the decoding, I'm not sure how we should handle the skipping of the `SetGroupVersionKind`. Thoughts?

https://github.com/aws/karpenter/blob/5392413df2fc6963ea1f80a6cc88970c73abcda2/pkg/cloudprovider/aws/apis/v1alpha1/provider.go#L54-L67

Reply:
If it's nil, I think we can just leave it fully empty. We can just return an empty constraints object for now. I haven't fully thought it through yet, but this is one of my least favorite pieces of our API. I think we will likely separate it out into a new CRD and make `provider` into `providerRef` or something.